### PR TITLE
Makes extinguishers not hit people when using help intent, improves code of backpack firefighter tank

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
Semi-revival of #10959, trying to extinguish someone while on help intent while adjacent to them will now not attack them. It also makes the the backpack firefighter tank more reliable, have better grammar, and removes some sprites with floating decals. It also fixes several weird things(i.e. the atmos backpack nozzle being a child of a mini fire extinguisher) in fire extinguisher code and makes it more readable.

Also, with the implementation of #18258, its more important that you don't hit yourself with fire extinguishers, which this stops.

## Why It's Good For The Game
As funny as it is to bash your coworker over the head with a fire extinguisher, it is very annoying. You have to purposely take a step back and hope they dont run away or click on the tile under them. 

## Changelog
:cl: Contrabang, Farie82
tweak: Fire extinguishers will now put out people on help intent instead of bashing them with it.
tweak: You can no longer bash yourself with a fire extinguisher when its open and on help intent.
tweak: The backpack firefighter tank is now more reliable and has better feedback.
tweak: The backpack firefighter tank's nozzle can now be dropped and will automatically return to the tank.
spellcheck: The backpack firefighter tank's grammar has been updated to better.
imagedel: The backpack firefighter tank no longer has floating icons for what mode its in.
/:cl:

### CODE REVIEW

**Comment by Contrabang** [comment]
Its failing CI here but working on my local server, I will look into it more later when I can

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 1]
```diff
@@ -248,53 +258,55 @@
 		loc = tank
 	return
 
-/obj/item/extinguisher/mini/nozzle/Move()
+/obj/item/extinguisher/nozzle/Move()
 	..()
 	if(tank && loc != tank.loc)
 		loc = tank
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/tanks/watertank.dm:264]: this entire proc worries me, NODROP may just be a better option here.

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 2]
```diff
@@ -248,53 +258,55 @@
...
 	return
 
-/obj/item/extinguisher/mini/nozzle/attack_self(mob/user as mob)
+obj/item/extinguisher/nozzle/attack(mob/living/M, mob/living/user, def_zone)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/tanks/watertank.dm:267]:
```suggestion
/obj/item/extinguisher/nozzle/attack(mob/living/M, mob/living/user, def_zone)
```

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 3]
```diff
@@ -248,53 +258,55 @@
...
 			return
 	return
 
-/obj/item/extinguisher/mini/nozzle/dropped(mob/user as mob)
+/obj/item/extinguisher/nozzle/dropped(mob/user as mob)
 	..()
+	nozzle_mode = EXTINGUISHER // dropping the nozzle in any other mode makes it not reliably be created again
 	to_chat(user, "<span class='notice'>The nozzle snaps back onto the tank!</span>")
 	tank.on = FALSE
 	loc = tank
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/tanks/watertank.dm:293]: what if the tank is null?

File: code/game/objects/items/weapons/tanks/watertank.dm (hunk 1 above)
Comment (Contrabang) [reply code/game/objects/items/weapons/tanks/watertank.dm:264]: I don't think this proc is actually required as the new version should automatically set its location to the tank, addressing if there is no tank can be handled in the dropped proc too

**Comment by hal9000PR** [comment]
2 design team objections and nobody in favour. closing.

**Pull request closed without merging**